### PR TITLE
 [FLINK-35200][table] Support the execution of suspend, resume materialized table in full refresh mode

### PR DESCRIPTION
**Pull request by hackergin**
## What is the purpose of the change

*Support the execution of suspend, resume materialized table in full refresh mode*

## Brief change log

* Add dynamic options for ResumeRefreshWorkflow interface.
*  Support the execution of suspend, resume materialized table in full refresh mode 

## Verifying this change

* Add test case testAlterMaterializedSuspendAndResumeInFullMode in MaterializedTableStatementITCase to verify the suspend and resume operation for materialized table in full mode.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (will be added in another pr)

### CODE REVIEW

**Comment by hackergin** [comment]
@flinkbot run azure

File: flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java [hunk 1]
```diff
@@ -652,6 +654,158 @@ void testAlterMaterializedTableWithoutSavepointDirConfigured() throws Exception
...
+        WorkflowInfo workflowInfo =
+                fromJson((String) jobDetail.getJobDataMap().get(WORKFLOW_INFO), WorkflowInfo.class);
+        assertThat(workflowInfo.getDynamicOptions())
+                .containsEntry("debezium-json.ignore-parse-errors", "true");
+
+        // delete the workflow
+        embeddedWorkflowScheduler.deleteScheduleWorkflow(jobKey.getName(), jobKey.getGroup());
+    }
+
+    private void createMaterializedTableInFullMode(String tableName, List<Row> data)
```
Comment (lsyldliu) [inline flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java:760]: Why don't you reuse the util method `createAndVerifyCreateMaterializedTableWithData`?

File: flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java [hunk 2]
```diff
@@ -652,6 +654,158 @@ void testAlterMaterializedTableWithoutSavepointDirConfigured() throws Exception
...
+                EmbeddedRefreshHandlerSerializer.INSTANCE.deserialize(
+                        refreshHandler, getClass().getClassLoader());
+
+        String workflowName = suspendRefreshHandler.getWorkflowName();
+        String workflowGroup = suspendRefreshHandler.getWorkflowGroup();
+        EmbeddedQuartzScheduler embeddedWorkflowScheduler =
+                SQL_GATEWAY_REST_ENDPOINT_EXTENSION
+                        .getSqlGatewayRestEndpoint()
+                        .getQuartzScheduler();
+        JobKey jobKey = new JobKey(workflowName, workflowGroup);
```
Comment (lsyldliu) [inline flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java:705]:
```suggestion
        JobKey jobKey = JobKey.jobKey(workflowName, workflowGroup);
```

File: flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java [hunk 3]
```diff
@@ -652,6 +654,158 @@ void testAlterMaterializedTableWithoutSavepointDirConfigured() throws Exception
...
+        Trigger.TriggerState suspendTriggerState =
+                embeddedWorkflowScheduler
+                        .getQuartzScheduler()
+                        .getTriggerState(new TriggerKey(workflowName, workflowGroup));
```
Comment (lsyldliu) [inline flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java:709]: TriggerKey.triggerKey(workflowName, workflowGroup)

File: flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java [hunk 4]
```diff
@@ -652,6 +654,158 @@ void testAlterMaterializedTableWithoutSavepointDirConfigured() throws Exception
...
+                        refreshHandler, getClass().getClassLoader());
+
+        assertThat(resumeRefreshHandler.getWorkflowName()).isEqualTo(workflowName);
+        assertThat(resumeRefreshHandler.getWorkflowGroup()).isEqualTo(workflowGroup);
+
+        JobDetail jobDetail = embeddedWorkflowScheduler.getQuartzScheduler().getJobDetail(jobKey);
+        Trigger.TriggerState resumedTriggerState =
+                embeddedWorkflowScheduler
+                        .getQuartzScheduler()
+                        .getTriggerState(new TriggerKey(workflowName, workflowGroup));
```
Comment (lsyldliu) [inline flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java:748]: ditto

**Review by lsyldliu — COMMENTED** [review]
@hackergin Thanks for contribution, I left some comments.

**Comment by hackergin** [comment]
@flinkbot run azure

File: flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java (hunk 1 above)
Comment (hackergin) [reply flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/MaterializedTableStatementITCase.java]: createAndVerifyCreateMaterializedTableWithData creates a table in continous mode, I made some modifications to support full mode.

**Review by lsyldliu — APPROVED** [review]
LGTM

**Pull request closed without merging**